### PR TITLE
Ros2 v0.8.0 lane departure checker

### PR DESCRIPTION
**Pull request by wep21**
blocks https://github.com/tier4/autoware_launcher.iv.universe/pull/44

### CODE REVIEW

File: control/trajectory_follower/lane_departure_checker/src/lane_departure_checker_node/lane_departure_checker_node.cpp [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+
+  output_ = lane_departure_checker_->update(input_);
+
+  updater_.force_update();
+
+  {
+    const auto & deviation = output_.trajectory_deviation;
+    debug_publisher_.publish<std_msgs::msg::Float64>("deviation/lateral", deviation.lateral);
+    debug_publisher_.publish<std_msgs::msg::Float64>("deviation/yaw", deviation.yaw);
+    debug_publisher_.publish<std_msgs::msg::Float64>("deviation/yaw_deg", rad2deg(deviation.yaw));
```
Comment (s-azumi) [inline control/trajectory_follower/lane_departure_checker/src/lane_departure_checker_node/lane_departure_checker_node.cpp:321]: std_msgs is deprecated

File: control/trajectory_follower/lane_departure_checker/package.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  <depend>autoware_planning_msgs</depend>
+  <depend>autoware_utils</depend>
+  <depend>boost</depend>
+  <depend>diagnostic_updater</depend>
+  <depend>eigen</depend>
+  <depend>geometry_msgs</depend>
+  <depend>lanelet2_extension</depend>
+  <depend>rclcpp</depend>
+  <depend>std_msgs</depend>
```
Comment (s-azumi) [inline control/trajectory_follower/lane_departure_checker/package.xml:19]: std_msgs is deprecated

File: control/trajectory_follower/lane_departure_checker/include/lane_departure_checker/lane_departure_checker_node.hpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+#include "autoware_utils/ros/self_pose_listener.hpp"
+#include "diagnostic_updater/diagnostic_updater.hpp"
+#include "lanelet2_core/LaneletMap.h"
+#include "lanelet2_extension/utility/message_conversion.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "autoware_planning_msgs/msg/route.hpp"
+#include "autoware_planning_msgs/msg/trajectory.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "std_msgs/msg/float64.hpp"
```
Comment (s-azumi) [inline control/trajectory_follower/lane_departure_checker/include/lane_departure_checker/lane_departure_checker_node.hpp:32]: std_msgs is deprecated.
Can you use autoware_debug_msgs/msg/Float64Stamped instead?

**Review by s-azumi — CHANGES_REQUESTED** [review]

File: control/trajectory_follower/lane_departure_checker/src/lane_departure_checker_node/lane_departure_checker_node.cpp (hunk 1 above)
Comment (wep21) [reply control/trajectory_follower/lane_departure_checker/src/lane_departure_checker_node/lane_departure_checker_node.cpp]: fixed.

File: control/trajectory_follower/lane_departure_checker/package.xml (hunk 2 above)
Comment (wep21) [reply control/trajectory_follower/lane_departure_checker/package.xml]: fixed

File: control/trajectory_follower/lane_departure_checker/include/lane_departure_checker/lane_departure_checker_node.hpp (hunk 3 above)
Comment (wep21) [reply control/trajectory_follower/lane_departure_checker/include/lane_departure_checker/lane_departure_checker_node.hpp]: fixed

File: common/util/autoware_utils/include/autoware_utils/ros/debug_traits.hpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+#include "autoware_debug_msgs/msg/float32_stamped.hpp"
+#include "autoware_debug_msgs/msg/float64_multi_array_stamped.hpp"
+#include "autoware_debug_msgs/msg/float64_stamped.hpp"
+#include "autoware_debug_msgs/msg/int32_multi_array_stamped.hpp"
+#include "autoware_debug_msgs/msg/int32_stamped.hpp"
+#include "autoware_debug_msgs/msg/int64_multi_array_stamped.hpp"
+#include "autoware_debug_msgs/msg/int64_stamped.hpp"
+#include "autoware_debug_msgs/msg/string_stamped.hpp"
+
+namespace debug_traits
```
Comment (kenji-miyake) [inline common/util/autoware_utils/include/autoware_utils/ros/debug_traits.hpp:31]: @wep21 Thank you for creating this library. I think it's better to add namespace of package name(autoware_utils).

Comment (wep21) [reply common/util/autoware_utils/include/autoware_utils/ros/debug_traits.hpp:33]: Thanks for your comment. I fixed it.

**Review by kenji-miyake — APPROVED** [review]

**Review by s-azumi — APPROVED** [review]

**Pull request merged**